src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
If we think it makes to support wrapX for non-global projections (with an extent) then it makes sense to make wrapX default to false. This would be my preference, because I don't see why wrapX should be reserved to projections covering the earth.